### PR TITLE
rats-tls: use VERSION file as the package version when running make p…

### PR DESCRIPTION
**Pull request by HuangYijuan**
…ackage

Fixes: #1331
Signed-off-by: Yijuan Huang <yijuan.huang@intel.com>

### CODE REVIEW

File: rats-tls/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Root directory of the project (absolute path).
+ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+RATS_TLS_VERSION := $(shell cat $(ROOTDIR)/VERSION)
+RATS_TLS_MAINTAINER := $(shell head -1 $(ROOTDIR)/MAINTAINERS)
+
+clean:
+	@rm -rf /usr/local/lib/rats-tls
```
Comment (haosanzi) [inline rats-tls/Makefile:7]: When typing `make clean` command, `make -C build clean` should executed.

File: rats-tls/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	@rm -f dist/rpm/rats_tls.spec dist/deb/debian/changelog
+
+install:
+	cmake -DBUILD_SAMPLES=on -H. -Bbuild
+	make -C build install
+
+uninstall:
+	@rm -rf /usr/local/include/rats-tls/
```
Comment (haosanzi) [inline rats-tls/Makefile:15]: Please use global env (such as RATS_LTB) to refer to `/usr/local/include/rats-tls/`

File: rats-tls/dist/deb/debian/rules [hunk 3]
```diff
@@ -7,10 +7,12 @@ BUILD_ROOT := $(CURDIR)/debian/rats-tls
...
 
 override_dh_auto_build:
-	make -C rats-tls SGX=1
+	cd rats-tls && cmake -H. -Bbuild
 
 override_dh_auto_install:
-	Rats_Tls_Root=$(BUILD_ROOT)/opt/rats-tls Rats_Tls_Bindir=$(BUILD_ROOT)/usr/share/rats-tls/samples make -C rats-tls install
+	cd rats-tls && make -C build DESTDIR=$(BUILD_ROOT) install
+
+override_dh_usrlocal:
```
Comment (haosanzi) [inline rats-tls/dist/deb/debian/rules:18]: Please type command `dpkg -r rats_tls` to see if `/usr/share/rats-tls/` and ` /usr/local/include/rats-tls/`, `/usr/local/lib/rats-tls` are removed.
If not, maybe you should add `postrm` file in Debian. Please refer to the [link\(https://github.com/alibaba/inclavare-containers/blob/master/shim/dist/deb/debian/postrm) for more information.

Comment (HuangYijuan) [reply rats-tls/dist/deb/debian/rules:18]: I've already tested dpkg -r rats-tls, they are all removed

File: rats-tls/Makefile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Root directory of the project (absolute path).
+ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+RATS_TLS_VERSION := $(shell cat $(ROOTDIR)/VERSION)
+RATS_TLS_MAINTAINER := $(shell head -1 $(ROOTDIR)/MAINTAINERS)
+Rats_Tls_Libdir := /usr/local/lib/rats-tls
+Rats_Tls_Incdir := /usr/local/include/rats-tls
+Rats_Tls_Bindir:= /usr/share/rats-tls
+
```
Comment (haosanzi) [inline rats-tls/Makefile:8]: Please add the rule for target `all: `

File: rats-tls/Makefile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+install:
+	cmake -DBUILD_SAMPLES=on -H. -Bbuild
+	make -C build install
+
+uninstall:
+	@rm -rf $(Rats_Tls_Libdir) $(Rats_Tls_Incdir) $(Rats_Tls_Bindir)
+
+package:
+	$(MAKE) -C dist package RATS_TLS_VERSION="$(RATS_TLS_VERSION)" RATS_TLS_MAINTAINER="$(RATS_TLS_MAINTAINER)"
+
```
Comment (haosanzi) [inline rats-tls/Makefile:22]: Please delete this empty line

**Review by haosanzi — APPROVED** [review]

File: rats-tls/dist/rpm/rats_tls.spec.in [hunk 6]
```diff
@@ -61,22 +52,21 @@ if [ ! -d "$SGX_SDK" ]; then
 fi
 
 pushd %{name}
-make SGX=1
+cmake -H. -Bbuild
```
Comment (zhiminghufighting) [inline rats-tls/dist/rpm/rats_tls.spec.in:55]: i suggest we can input a parameter here to specify the build mode such as "sgx" "tdx" or "occlum" etc.

**Review by zhiminghufighting — APPROVED** [review]

**Pull request merged**